core/src/BitMatrix.cpp

```diff
@@ -96,15 +97,7 @@ BitMatrix::setRegion(int left, int top, int width, int height)
 void
 BitMatrix::rotate180()
 {
-	BitArray topRow(_width), bottomRow(_width);
-	for (int i = 0; i < (_height + 1) / 2; i++) {
-		getRow(i, topRow);
-		getRow(_height - 1 - i, bottomRow);
-		topRow.reverse();
-		bottomRow.reverse();
-		setRow(i, bottomRow);
-		setRow(_height - 1 - i, topRow);
-	}
+	BitHacks::Reverse(_bits, _rowSize * 32 - _width);
```

Comment:
It does indeed work. E.g.:

> 1010....0011....

becomes

> ....1100....0101

if reversed and then needs to be shifted by 4 places, becomes:

> 1100....0101....

So the gaps between the rows just need to be shifted as well.